gravitee-apim-integration-tests/src/test/java/io/gravitee/apim/integration/tests/tcp/TcpGatewayWebsocketIntegrationTest.java

```diff
@@ -0,0 +1,116 @@
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.rxjava3.core.http.HttpClient;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Benoit BORDIGONI (benoit.bordigoni at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@GatewayTest
+class TcpGatewayWebsocketIntegrationTest extends AbstractWebsocketGatewayTest {
+
+    @Override
+    public void configureEntrypoints(Map<String, EntrypointConnectorPlugin<?, ?>> entrypoints) {
+        entrypoints.putIfAbsent("tcp-proxy", EntrypointBuilder.build("tcp-proxy", TcpProxyEntrypointConnectorFactory.class));
+    }
+
+    @Override
+    public void configureEndpoints(Map<String, EndpointConnectorPlugin<?, ?>> endpoints) {
+        endpoints.putIfAbsent("tcp-proxy", EndpointBuilder.build("tcp-proxy", TcpProxyEndpointConnectorFactory.class));
+    }
+
+    @Override
+    protected void configureGateway(GatewayConfigurationBuilder gatewayConfigurationBuilder) {
+        super.configureGateway(gatewayConfigurationBuilder);
+        // enables the TCP proxy
+        gatewayConfigurationBuilder.configureTcpGateway(tcpPort());
+    }
+
+    @Override
+    protected void configureHttpClient(HttpClientOptions options) {
+        // we are doing a websocket anyway, so we need an HTTP Client, but we use the TCP port
+        options.setDefaultHost("localhost").setDefaultPort(tcpPort()).setForceSni(true).setSsl(true).setVerifyHost(false).setTrustAll(true);
+    }
+
+    @Test
+    @DeployApi({ "/apis/v4/tcp/api.json" })
+    void should_call_websocket_backend_via_TCP_port(VertxTestContext testContext, HttpClient httpClient) {
+        var serverConnected = testContext.checkpoint();
+        var serverMessageSent = testContext.checkpoint();
+        var serverMessageChecked = testContext.checkpoint();
+
+        // simple websocket answering PONG when sent "PING"
+        websocketServerHandler =
+            serverWebSocket -> {
+                serverConnected.flag();
+                serverWebSocket.exceptionHandler(testContext::failNow);
+                serverWebSocket.accept();
+                serverWebSocket.frameHandler(frame -> {
+                    if (frame.isText()) {
+                        testContext.verify(() -> assertThat(frame.textData()).isEqualTo("PING"));
+                        serverWebSocket
+                            .writeTextMessage("PONG")
+                            .doOnComplete(serverMessageSent::flag)
+                            .doOnError(testContext::failNow)
+                            .subscribe();
+                    }
+                });
+            };
+
+        // Given that a websocket is deployed
+        // When calling the TCP proxy with an Api point ing to the websocket
+        // Then we should expect a normal bidi websocket call
+        httpClient
+            .webSocket("/test")
```

Comment:
do we agree that `/test` path means nothing here and could be whatever ?
We are calling the api because of the host "localhost" ?